packages/react-checkbox/src/components/Checkbox/useCheckbox.tsx

```diff
@@ -0,0 +1,112 @@
+import * as React from 'react';
+import {
+  makeMergeProps,
+  resolveShorthandProps,
+  useControllableState,
+  useId,
+  useIsomorphicLayoutEffect,
+  useMergedRefs,
+} from '@fluentui/react-utilities';
+import { CheckboxProps, CheckboxShorthandProps, CheckboxState } from './Checkbox.types';
+import { Label } from '@fluentui/react-label';
+import { DefaultCheckmarkIcon, DefaultMixedIcon } from './DefaultIcons';
+
+/**
+ * Array of all shorthand properties listed in CheckboxShorthandProps
+ */
+export const checkboxShorthandProps: CheckboxShorthandProps[] = ['label', 'indicator', 'input'];
+
+const mergeProps = makeMergeProps<CheckboxState>({ deepMerge: checkboxShorthandProps });
+
+/**
+ * Create the state required to render Checkbox.
+ *
+ * The returned state can be modified with hooks such as useCheckboxStyles,
+ * before being passed to renderCheckbox.
+ *
+ * @param props - props from this instance of Checkbox
+ * @param ref - reference to root HTMLElement of Checkbox
+ * @param defaultProps - (optional) default prop values provided by the implementing type
+ */
+export const useCheckbox = (
+  props: CheckboxProps,
+  ref: React.Ref<HTMLElement>,
+  defaultProps?: CheckboxProps,
+): CheckboxState => {
+  const state = mergeProps(
+    {
+      ref,
+      id: useId('checkbox-'),
```

Comment:
Even if it works the proper way to is to do reassignment as `useId()` handles it:

```diff
  const state = mergeProps(
    {
      ref,
-      id: useId('checkbox-'),
//

+  state.id = useId('checkbox-', props.id),
```

It will become more important when we will need to complete #18844.